src/misc/JsonConfig.cpp

```diff
@@ -0,0 +1,173 @@
+
+JsonConfig::JsonConfig(std::string path) : mPath(path) {
+    // check if path exists and that it's a file
+    if (!fs::exists(path) || !fs::is_regular_file(path)) {
+        mIsNewConfig = true;
+        mJson = json::object();
+        return;
+    }
+
+    try {
+        std::ifstream ifs(path);
+        mJson = json::parse(ifs);
+    } catch (json::parse_error& e) {
+        SPDLOG_ERROR("Failed to parse JSON file: {}", e.what());
+
+        // If failure to parse, we will create a new file
+        mIsNewConfig = true;
+        mJson = json::object();
+    }
+}
+
+json::json_pointer JsonConfig::DotKeyToPointer(std::string key) {
+    std::replace(key.begin(), key.end(), '.', '/');
+    key = "/" + key;
+
+    return json::json_pointer(key);
+}
+
+void JsonConfig::SetArbitraryType(std::string key, json::value_type value) {
+    auto ptr = DotKeyToPointer(key);
+    mJson[ptr] = value;
+}
+
+json::value_type JsonConfig::GetArbitraryType(std::string key) {
+    auto ptr = DotKeyToPointer(key);
+
+    try {
+        return mJson.at(ptr);
+    } catch (json::out_of_range& e) { return nullptr; }
+}
+
+// MARK: - Public API
+
+bool JsonConfig::IsNewConfig() {
+    return mIsNewConfig;
+}
+
+void JsonConfig::DeleteEntry(std::string key) {
+    // nlohmann::json doesn't have a way to delete a json pointer so
+    // we'll work around it by grabbing the parent object of the key
+    // we want to delete and deleting the key
+
+    auto currentKey = key;
+    auto parentKey = key;
+
+    try {
+        bool shouldStop = false;
+        while (!shouldStop) {
+            // grab parent of current key
+            parentKey = currentKey.substr(0, currentKey.find_last_of('.'));
+            auto parentPtr = DotKeyToPointer(parentKey);
+
+            // grab parent object at key and delete the current key
+            auto& parent = mJson.at(parentPtr);
+            auto keyToDelete = currentKey.substr(currentKey.find_last_of('.') + 1);
+
+            // check if parent object is an array and delete index
+            if (parent.is_array()) {
+                auto index = std::stoi(keyToDelete);
+                parent.erase(index);
+            } else {
+                parent.erase(keyToDelete);
+            }
+
+            // set variable for whether we should keep recursing parents
+            shouldStop = parent.empty() || parentKey == currentKey;
+            currentKey = parentKey;
+        }
+    } catch (json::out_of_range& e) {
+        // at can throw, showing that a parent object doesn't exist
+        // ignore and don't do anything
+    }
+}
+
+json::value_type JsonConfig::GetRawEntry(std::string key) {
+    return GetArbitraryType(key);
+}
+
+void JsonConfig::SetInteger(std::string key, int32_t value) {
+    return SetArbitraryType(key, value);
+}
+
+int32_t JsonConfig::GetInteger(std::string key, int32_t defaultValue) {
+    json::value_type value = GetArbitraryType(key);
+    if (value && value.is_number_integer()) {
```

Comment:
For here and all the ones below, seems like you need to explicitly check against nullptr, as just checking itself as truthy when it is a valid json value is actually executing `.get<false>()` behind the scenes, which will throw for any non boolean types.
```suggestion
    if (value != nullptr && value.is_number_integer()) {
```